app/javascript/packages/analytics/index.spec.ts

```diff
@@ -26,44 +40,139 @@ describe('trackEvent', () => {
     });
 
     context('no payload', () => {
-      it('fetches and resolves to undefined', async () => {
+      it('calls sendBeacon and resolves to undefined', async () => {
         const result = await trackEvent('name');
 
         expect(result).to.be.undefined();
-        expect(global.fetch).to.have.been.calledWith(
-          endpoint,
-          sandbox.match({
-            body: '{"event":"name"}',
-            headers: { 'Content-Type': 'application/json' },
-            method: 'POST',
-          }),
-        );
+
+        expect(global.navigator.sendBeacon).to.have.been.calledOnce();
+
+        const [actualEndpoint, data] = (global.navigator.sendBeacon as SinonStub).firstCall.args;
+        expect(actualEndpoint).to.eql(endpoint);
+        expect(data).to.have.property('type').eql('application/json');
+
+        expect(await blobTextContents(data)).to.eql('{"event":"name"}');
```

Comment:
I could not find a more pleasant way of verifying the contents of a `Blob` with sinon.

Reply:
I might be getting the types mixed up, but might be able to use/copy the `isArrayBufferEqual` utility function recently removed in #8396? At least if comparing the raw data might be preferable. Not aware of an option that doesn't involve a utility function though.

https://github.com/18F/identity-idp/blob/6447b1e65a4a297c85877309edcd06835f03d457/spec/javascript/packages/document-capture/higher-order/with-background-encrypted-upload-spec.jsx#L13-L33